core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -275,7 +275,7 @@ public boolean allocatedPostIndexCreate(IndexMetaData indexMetaData) {
     }
 
     public boolean isPeerRecovery() {
-        return !primary() || relocatingNodeId() != null;
+        return state == ShardRoutingState.INITIALIZING && (primary() == false || relocatingNodeId != null);
```

Comment:
with the change to only consider initializing shards, I agree - it's more explicit and better. can we add java docs to the method that it only returns true for intializing shards  that should pull their data from a remote copy?